app/plugin_api.go

```diff
@@ -369,6 +369,39 @@ func (api *PluginAPI) GetUserCustomStatus(userID string) (*model.CustomStatus, *
 	return api.app.GetCustomStatus(userID)
 }
 
+func (api *PluginAPI) RestoreToPreviousCustomStatus(userID string) *model.AppError {
+	cs, appErr := api.app.GetCustomStatus(userID)
+	if appErr != nil {
+		return appErr
+	}
+
+	// If user changed the custom status explicitly, don't restore to previous status
+	if cs == nil || !cs.SetByProduct {
+		return model.NewAppError("RestoreToPreviousCustomStatus", "plugin.api.restore_to_previous_custom_status", nil, "", http.StatusBadRequest)
+	}
```

Comment:
Is this a real error case? Or, we are just ignoring the restoration process? If the latter, I don't think it's necessary to return error.

Reply:
its the latter case. Changed it to return `nil`